widgetry/src/widgets/line_plot.rs

```diff
@@ -225,15 +186,15 @@ impl<T: Yvalue<T>> WidgetImpl for LinePlot<T> {
                     Pt2D::new(cursor.x - self.top_left.x, cursor.y - self.top_left.y),
                     radius,
                 ) {
-                    // TODO If some/all of the matches have the same t, write it once?
-                    let t = self.max_x.percent_of(pt.x() / self.dims.width);
+                    // TODO If some/all of the matches have the same x, write it once?
+                    let x = self.max_x.from_percent(pt.x() / self.dims.width);
                     let y_percent = 1.0 - (pt.y() / self.dims.height);
 
                     // TODO Draw this info in the ColorLegend
                     txt.add_line(format!(
                         "{}: at {}, {}",
                         label,
-                        t.ampm_tostring(),
+                        x.prettyprint(),
```

Comment:
One nit on the output - the x-axis origin label looks a little wonky:

<img width="242" alt="Screen Shot 2021-03-26 at 11 50 26 AM" src="https://user-images.githubusercontent.com/217057/112679503-77d23400-8e29-11eb-83df-cb6d63a861f0.png">

I think maybe partially because it's in the same units as the y-axis, the x-axis origin label almost looks like a broken y-axis label. It took my brain a bit to realize it's rotation was aligned to the other x-axis labels.
 
It's not a deal breaker for me - but if either are easy:

- maybe share a single "0" label for the x/y axis
- show "0 miles" instead of "0 feet" for the x-axis

I haven't looked, but I'm guessing both of these are non-trivial, so feel free to skip.

Reply:
Hmm... I'm going to punt on this for the moment. I'm not sure about sharing the 0 -- the x axis will always start at 0, but it's maybe a bit strange for the Y axis to do so. It compresses the shape of the plot if you start and end somewhere far above sea level.

"0 feet" -> "0 miles", and equivalently "0m" -> "0km", seems less strange to me. In most cases, the x axis is at least 0.1 miles, but rarely it could be under and we'd degenerate to showing feet there too.